syntax = "proto3";

package mock_message_board;

service MessageService {
  rpc findLastMessage (findLastMessageRequest) returns (findLastMessageReply);
  rpc sendMessage (saveMessageRequest) returns (saveMessageReply);
}

service SanitizationService {
  rpc sanitize_message (saveMessageRequest) returns (saveMessageReply);
}

service MockDatabase {
  rpc findLastMessage (findLastMessageRequest) returns (findLastMessageReply);
  rpc saveMessage (saveMessageRequest) returns (saveMessageRequest);
}

message saveMessageRequest {
  string client_id = 1;
  string message = 2;
  uint64 query_uid = 3;
}

message saveMessageReply {
  bool ok = 1;
  uint64 query_uid = 2;
}

message findLastMessageRequest {
  string client_id = 1;
  uint64 query_uid = 2;
}

message findLastMessageReply {
  string message = 1;
  uint64 query_uid = 2;
}
